tests/run/cdef_class_order.pyx

```diff
@@ -5,11 +5,41 @@ cdef class B
 cdef class A(object):
     cdef list dealloc1
 
+cdef class Y(X): pass
+cdef class X(C): pass
+cdef class C: pass
+
 cdef class B(A):
     cdef list dealloc2
 
+cdef class Z(A): pass
+
+
 def test():
     """
     >>> test()
+    A
+    B
+    C
+    X
+    Y
+    Z
     """
-    A(), B()
+    A(), B(), C(), X(), Y(), Z()
+    import sys
+    py_ver = sys.version_info[:2]
+    py_impl = getattr(sys, 'implementation', None)
+    dicts_are_ordered = (
+        py_ver >= (3, 7)
+        or (
+            py_ver == (3, 6)
+            and py_impl.name == 'cpython'
```

Comment:
Personally I'd be tempted just to make the test `py_ver >= (3, 7)` and not bother with the 3.6 on cpython part